backend/src/Designer/Services/Implementation/SchemaModelService.cs

```diff
@@ -431,5 +435,20 @@ private bool NamespaceNeedsToBeSeparated(ApplicationMetadata application,
         {
             return application.DataTypes.All(d => d.AppLogic?.ClassRef != $"Altinn.App.Models.{csharpModelName}");
         }
+
+        public async Task<DataType> GetModelDataType(string org, string app, string modelId)
+        {
+            ApplicationMetadata applicationMetadata = await _applicationMetadataService.GetApplicationMetadataFromRepository(org, app);
+            DataType dataType = applicationMetadata.DataTypes.Find((dataType) => dataType.Id == modelId);
+            return dataType;
+        }
+
+        public async Task SetModelDataType(string org, string app, string modelId, DataType dataType)
+        {
+            ApplicationMetadata applicationMetadata = await _applicationMetadataService.GetApplicationMetadataFromRepository(org, app);
+            applicationMetadata.DataTypes.RemoveAll((dataType) => dataType.Id == modelId);
+            applicationMetadata.DataTypes.Add(dataType);
+            await _applicationMetadataService.UpdateApplicationMetaDataLocally(org, app, applicationMetadata);
```

Comment:
Two questions regarding this method: 
1. Can you change the dataTypeId to anything with this endpoint? The new Id must be validated towards the other datatypeIds.
2. We should be certain that the IDs are not duplicated in the dataTypes, can we use `Remove`?

Reply:
> Can you change the dataTypeId to anything with this endpoint? The new Id must be validated towards the other datatypeIds.

I reworked this to not use the `modelId` parameter as it doesn't seem necessary. Is the logic more clear now?

> We should be certain that the IDs are not duplicated in the dataTypes, can we use Remove?

Can you elaborate on what you mean by using `Remove`?

Reply:
> I reworked this to not use the modelId parameter as it doesn't seem necessary. Is the logic more clear now?

Looks better! 

> Can you elaborate on what you mean by using Remove?

Not `RemoveAll` but only `Remove`. Does not make a huge difference, but it seems unnecessary traversing all DataTypes here if the first dataType is the one we wish to update.

Reply:
If we keep `RemoveAll` it will also be able to act as a contingency in case this list somehow does have multiple objects with the same ID. And given the list is unlikely to be any bigger than a several items i don't think it will cost much to iterate a bit more. Do you agree?

Reply:
Personally I would go for `Remove` just to clearify that this list should never have multiple Ids, but as you say, it probably wont affect performance particularly so we can keep as is 👍